diff --color -Naur cf-units-3.2.0.orig/cf_units/__init__.py cf-units-3.2.0/cf_units/__init__.py
--- cf-units-3.2.0.orig/cf_units/__init__.py	2023-04-17 17:11:11.000000000 +0200
+++ cf-units-3.2.0/cf_units/__init__.py	2024-05-01 21:13:14.077625410 +0200
@@ -1847,7 +1847,7 @@
                 )
                 convert_type = isinstance(
                     value, np.ndarray
-                ) and np.issubsctype(value.dtype, np.floating)
+                ) and np.issubdtype(value.dtype, np.floating)
                 if convert_type:
                     result = result.astype(value.dtype)
             else:
diff --color -Naur cf-units-3.2.0.orig/cf_units/tests/test_unit.py cf-units-3.2.0/cf_units/tests/test_unit.py
--- cf-units-3.2.0.orig/cf_units/tests/test_unit.py	2023-04-17 17:11:11.000000000 +0200
+++ cf-units-3.2.0/cf_units/tests/test_unit.py	2024-05-01 21:40:05.267267536 +0200
@@ -914,7 +914,8 @@
         f = Unit("deg_f")
 
         # Manufacture a non-native byte-order array to be converted.
-        orig = np.arange(4, dtype=np.float32).newbyteorder().byteswap()
+        arr = np.arange(4, dtype=np.float32)
+        orig = arr.view(arr.dtype.newbyteorder("S"))
 
         emsg = (
             "Unable to convert non-native byte ordered array in-place. "
