diff --git a/setup.py b/setup.py
index de31178..706247f 100644
--- a/setup.py
+++ b/setup.py
@@ -74,7 +74,12 @@ def get_package_data():
             emsg = f"Require to set {xml_env} for a cf-units wheel build."
             raise ValueError(emsg)
         xml_database = Path(xml_database)
-        if not xml_database.is_file():
+        try:
+            is_file = xml_database.is_file()
+        except OSError:
+            # Handle Windows conda-forge build exception raised by pathlib.
+            return package_data
+        if not is_file:
             emsg = (
                 f"Can't open {xml_env} file {xml_database} "
                 "during cf-units wheel build."
